src/Sportalytics.Domain/Shared/Result.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Sportalytics.Domain.Shared;
+
+public class Result
+{
+    protected internal Result(bool isSuccess, Error error)
```

Comment:
Maybe you can use this class https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.results?view=aspnetcore-8.0 ?

Reply:
@Puchkof aspnetcore.http.results focus on HTTP results. I want to have my own result class where I can handle the result status in my own service or between services where I will not use HTTP and not depend on HTTP.
And aspnetcore.http.results don't have useful functions like IsSuccess or IsFailure